src/Hooks/GlobalMiddleware.php

```diff
@@ -31,10 +31,14 @@ public function __construct(
 
     public function handle(Request $request, Closure $next): mixed
     {
+        if ($this->hasHandledRequest) {
+            return $next($request);
+        }
+
+        $this->hasHandledRequest = true;
+
```

Comment:
This middleware was previously running twice when a route was matched. This boolean ensures we only run the handle method once.